### ISSUE TITLE
Table Formatting

### ISSUE THREAD

**Issue by n0s4** [artempyanykh/marksman#80]
Formatting tables manually is a pain. Does table formatting sound like a plausible feature? I can use a third party formatter in the meantime but it does sound convenient.

**Comment by artempyanykh** [comment]
@n0s4 yes, as a code action it makes a lot of sense!

**Comment by David-Else** [comment]
I hope marksman does not become a tool for formatting, we have prettier for that. If that is too slow then there is Deno written in rust that is very fast.

If it does become a formatter then please make sure there is as an option to turn it off. Cheers :)

**Comment by artempyanykh** [comment]
To set clear expectations, unless someone comes, implements this code action and sends a PR (which I'd be happy to accept because I do see value in this functionality), I wouldn't count on Marksman becoming a formatter any time soon - my personal todo list is already quite full with improvements to the *core* functionality.

**Comment by artempyanykh** [comment]
@jfaz1 I agree it'd be convenient! In terms of UX, `org-mode` tables come to mind first. But I do wonder what it would look like in terms of LSP interaction. The protocol doesn't allow much dynamism, so I don't really know how a code action (or anything else) would support the use-case of dynamic table creation/manipulation.

**Comment by rlees85** [comment]
Disagree with some comments. LSP supports and should handle formatting. It would be great if marksman can format the markdown for you the same as other LSP does on-save (such as gopls, terraform-ls, etc). If it does not yet do so already (I ended up here seeing if it can rather than to complain that it can't).

**Comment by chrisgrieser** [comment]
Just wanna add that updating the ToC is something that I can see being part of the formatting rather than a code action.

**Comment by savetheclocktower** [comment]
[md-lsp](https://github.com/matkrin/md-lsp) does reformatting of tables as part of its code-formatting support; maybe someone could find inspiration in how it implements code formatting. It's the one feature that keeps me using `md-lsp` occasionally instead of Marksman all the time.

(I'd consider a contribution, but F# is a bit outside of my wheelhouse.)

**Comment by jolars** [comment]
For anyone who's interested in this, I have made a formatter (and LSP and linter) for Pandoc markdown at https://github.com/jolars/panache, which support formatting of simple, pipe, grid, and multiline tables.

**Comment by Freed-Wu** [comment]
> a Create a Table code action 

before it is finished, some plugin like [table-mode](http://github.com/dhruvasagar/vim-table-mode) do it for me.

> Formatting tables

biome has a feature request to support markdown formatting https://biomejs.dev/internals/language-support/
